### DISCUSSION TITLE
1.0.9.9

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#91] [Обновления]
- Добавлена проверка на то, является ли переводимое видео Прямой трансляцией (youtube.com)
- Добавлена проверка на то, является ли переводимое видео Премьерой (youtube.com)
- Добавлена проверка на длину видео (youtube.com)

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.0.9.8...1.0.9.9

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.0.9.9'>1.0.9.9</a>.</em>

**Comment by FiLinX** [comment]
А можно поинтересоваться, вот это - "Добавлена проверка на длину видео (youtube.com)"  -  кем лимит выставляется? Возможно ли будет его самому правкой в скрипте увеличить?
Просто более половины переводимых роликов которые я смотрю - не мененее чем получасовые а ближе к часу, и перевод отлично работает! Ну если только иногда, на ещё прям только-только свежеиспечённых роликах - надо немножко (2-3 минуты) подождать перевода, и всё.

**Reply by ilyhalight** [reply]
> А можно поинтересоваться, вот это - "Добавлена проверка на длину видео (youtube.com)" - кем лимит выставляется? Возможно ли будет его самому правкой в скрипте увеличить?

Этим, я просто подогнал под лимит от Яндекса (видео должны быть менее 14400 секунд или же в более понятном формате - менее 4 часов). Сделал, так, дабы можно было быстрее понять, что такое видео не получится перевести и не посылать лишних запросов на сервера. Раньше запрос посылался на сервера Яндекса, но приходилось ждать некоторое время и получать отказ с ошибкой, что они не могут переводить такие длинные видео.

При желание это можно, конечно, самому убрать, просто, удалив эти строки из кода расширения:
```
if (video.duration > 14400) {
    throw "VOT: Видео слишком длинное";
}
```
 но смысла в этом я не вижу ¯\_(ツ)_/¯

**Comment by despro7** [comment]
Приветствую! Спасибо за ваш скрипт, очень крутой хелпер! Но у меня есть один баг, который немного достаёт, практически в каждом видео на Youtube слетают настройки громкости оригинала, я пока не смог определить закономерности, но громкость оригинала вдруг самопроизвольно меняется до 100%, и это немного подбешивает.

Также хотел предложить изменить UI/UX кнопки в плеере ютуба - разместить рядом с другими функциональными кнопками (субтитры, настройки) и выводить окошко настроек в правом нижнем углу, по примеру дизайна тех же настроек youtube-плеера. Думаю это будет гораздо удобнее, так текущий вариант размещения кнопки часто перекрывается оригинальным интерфейсом плеера. С радостью готов предложить свою помощь в вёрстке, если это актуально.

**Reply by ilyhalight** [reply]
> Но у меня есть один баг, который немного достаёт, практически в каждом видео на Youtube слетают настройки громкости оригинала, я пока не смог определить закономерности, но громкость оригинала вдруг самопроизвольно меняется до 100%, и это немного подбешивает.

Это не баг. Настройки громкости оригинала не сохраняются. Всем угодить невозможно - кому-то это нравится, кому-то наоборот. Если вам нужно установить определенную громкость видео с включенным переводом, то включите функцию: "Автоматически выставлять громкость оригинала (Yandex-like)" и в коде расширения установите нужное значение для переменной `defaultVideoVolume` (по умолчанию: `0.15` - 15%)

> Также хотел предложить изменить UI/UX кнопки в плеере ютуба - разместить рядом с другими функциональными кнопками (субтитры, настройки) и выводить окошко настроек в правом нижнем углу, по примеру дизайна тех же настроек youtube-плеера. Думаю это будет гораздо удобнее, так текущий вариант размещения кнопки часто перекрывается оригинальным интерфейсом плеера.

Возможно, это было бы удобнее, но, тогда, нужно будет менять дизайн и расположение кнопки и на других поддерживаемых сайтах. Если, вам, всё же, кажется, что так будет лучше - создайте топик в "Issue" и попросите других людей проголосовать "За" или "Против" вашего предложения - соберётся много голосов "За" - постараюсь реализовать.

**Comment by accountnujen** [comment]
Подскажите пожалуйста, как перевести видео, если у меня есть только видео, не загружая его никуда?

**Reply by ilyhalight** [reply]
Никак, для перевода в любом случае придётся куда-то загружать. Будь то ютуб или вимео

**Reply by accountnujen** [reply]
> Никак, для перевода в любом случае придётся куда-то загружать. Будь то ютуб или вимео

ок, другой подход тогда: видео загружено на моём сайте. Получить видео можно просто обратившись website.com/video.mp4 . Как заставить переводить на моём сайте?

**Reply by ilyhalight** [reply]
> > Никак, для перевода в любом случае придётся куда-то загружать. Будь то ютуб или вимео
> 
> ок, другой подход тогда: видео загружено на моём сайте. Получить видео можно просто обратившись website.com/video.mp4 . Как заставить переводить на моём сайте?

Тоже. никак, сам сервер яндекс переводчика поддерживает, только, заданный список сайтов. На других сайтах он просто выбивает оишбку

**Reply by accountnujen** [reply]
> только, заданный список сайтов

Дефолтный переводчик Яндекс Браузера не переводит порнхаб. Это уже, видимо, ваша доработка, раз он появился в этом скрипте.

**Reply by ilyhalight** [reply]
> > только, заданный список сайтов
> 
> Дефолтный переводчик Яндекс Браузера не переводит порнхаб. Это уже, видимо, ваша доработка, раз он появился в этом скрипте.

Всё, что указано в #34, все, те сайты, Яндекс оффициально поддерживает (или возможно, пока, только, планирует поддерживать, но эти сайты уже включены в список доступных на их сервере). Всё, что было сделано мной - просто адаптирован изначальный скрипт для работы с данными сайтами. На остальных сайтах, вы просто будете получать ошибку о том, что "Возникла ошибка, попробуйте позже" 
![изображение](https://user-images.githubusercontent.com/62353659/225922525-39bf5a51-c403-4cb7-a81d-72e550b4a7ac.png)


![изображение](https://user-images.githubusercontent.com/62353659/225912884-c2c8c6f7-254d-48fe-a02d-8ccc2d298de6.png)

**Reply by accountnujen** [reply]
> или возможно, пока, только, планирует поддерживать

и каким чудом вы выяснили, что pornhub они поддерживают?

**Reply by ilyhalight** [reply]
> > или возможно, пока, только, планирует поддерживать
> 
> и каким чудом вы выяснили, что pornhub они поддерживают?

Как сказал выше, из дампа Яндекс Браузера в #34

**Comment by imposible-Cxx** [comment]
Спасибо за скрипт! Протестировал в Vivaldi 5.7.2921.63 под Windows и Linux, на youtube работает озвучка.

**Reply by ilyhalight** [reply]
Спасибо за отзыв, чуть позже добавлю в список протестированных браузеров

**Comment by Degreyn** [comment]
Привет! Большое спасибо за такой хелпер!) Скажите, пожалуйста. Как я понял, он не может озвучивать видео с ограниченным доступом. Просто я раньше делал как, загружал курсы на английском языке на ютуб, он сам создавал субтитры на английском, а я уже соответственно переводил их на русский и так смотрел. А сейчас решил попробовать ваше расширение, но оно не хочет озвучивать, так как нет доступа к видео. А ограничение стоит, чтобы не заблокировали правообладатели. Возможно ли как-то сделать, что бы он научился получать доступ через ссылки например?) Чтобы не отрывать доступ к видео. Заранее, большое спасибо за ответ!)


пысы. И так, я изменил в настройках видео "доступ по ссылке" и о чудо, оно и так заработало) спасибо)

**Reply by ilyhalight** [reply]
В ограниченном доступе, только, автор ютуб канала может видеть ролики)

**Comment by warlockblack** [comment]
Спасибо большое, в Brave отлично работает. Но есть ли хоть какая-то возможность прикрутить Tampermonkey для Сафари вместо UserScripts? Дело в том, что UserScripts уже не работает с Mac OS Catalina, а возможности обновить систему пока нет.

**Reply by ilyhalight** [reply]
> Спасибо большое, в Brave отлично работает. Но есть ли хоть какая-то возможность прикрутить Tampermonkey для Сафари вместо UserScripts? Дело в том, что UserScripts уже не работает с Mac OS Catalina, а возможности обновить систему пока нет.

В теории, будет работать и в тамперманки на MacOS, попробуйте обе версии расширения

**Reply by ilyhalight** [reply]
> > Спасибо большое, в Brave отлично работает. Но есть ли хоть какая-то возможность прикрутить Tampermonkey для Сафари вместо UserScripts? Дело в том, что UserScripts уже не работает с Mac OS Catalina, а возможности обновить систему пока нет.
> 
> В теории, будет работать и в тамперманки на MacOS, попробуйте обе версии расширения

На практике, к сожалению ничего сказать не могу, т.к. не являюсь владельцем мака
